collection-scripts/gather_osus

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+source $(dirname "$0")/common.sh
+BASE_COLLECTION_PATH="must-gather"
+OSUS_OPERATOR_NAME="update-service-operator"
+get_log_collection_args
+
+HAS_OSUS=$(oc get csv -A --no-headers -o custom-columns=NS:.metadata.namespace,OPERATOR:.metadata.name --ignore-not-found=true | awk '/'${OSUS_OPERATOR_NAME}'/ {print $1}')
```

Comment:
This seems like a hard direction for an OCP-central must-gather to support.  I thought the OLM-installed operator pattern was for each operator to ship a separate image with the must-gather-for-them logic (as described in [the KCS][1] you'd linked from [OTA-1177](https://issues.redhat.com/browse/OTA-1177)), so the central must-gather maintainers didn't have to be bothered reviewing gather logic for all the many, many, OLM-installed operators that could possibly be present?

[1]: https://access.redhat.com/solutions/5459251

Reply:
@wking , while I agree with your comment for most of the operators that can be installed in OpenShift, I think creating a full image for this operator is excessive.
Other operators provide extra capacities to the cluster, and usually involve several different CRDs and even several namespaces. This operator helps with one of the core capacities of OpenShift, which is the upgrade (in this case, for disconnected clusters), and lots of cases already includes a must-gather proactively. The info for this operator (when installed) shouldn't increase a lot the size of the must-gather, and will avoid asking for a new different must-gather.

Reply:
@wking , any thoughts on the above?

Reply:
@oarribas / @wking the only thing this collects are the details from a given namespace[s]. I think this is fine (given its related to update); but @soltysh or @ingvagabund should have the final say if this is provided by this image or an independent image.

Reply:
@oarribas how big are the extra manifests? Can you share an example of running both commands?

Reply:
@ingvagabund , checking data from some cases, it depends a lot on the volume of the logs from the pods.
Largest inspect of the namespace I have seen, without compression is 30MB, and compressed between 2-3MB. The size of the `updateservice` resource is few KB.

Reply:
@ingvagabund , @soltysh , any thoughts based on the above?

Reply:
For some time we have been merging extra collection with a promise of collecting extra "small" data that help to avoid asking customers to run yet another must gather image. More relevant in the disconnected environment. Yet, we don't track how much of these extra collections increase the overall must-gather size on average. Before going further I'd like to see when all the extra collections are collected. E.g. what's the wort case extra bump in the collected data. A new section under https://github.com/openshift/must-gather/blob/master/README.md will do. E.g.

## Extra collections

| script location | short description       | condition | estimated size |
| ------------ | ----------------- | ----------- | ------ |
| /usr/bin/gather_aro | Gather ARO Cluster Data | `ns/openshift-azure-operator` or `ns/openshift-azure-logging` present | ?? |
| /usr/bin/gather_vsphere | Gather vSphere resources | vSphere CSI driver is installed | ?? |

Reply:
@ingvagabund , the estimated size for this one (for the OSUS operator), when compressed, is around 3MB (usually less as per several inspect of the `openshift-update-service` namespace I have reviewed). And it's only collected if OSUS is installed.